R/fileupload.R

```diff
@@ -52,7 +78,8 @@ FileUploadOperation <- R6Class(
       .currentFileInfo <<- file
       .pendingFileInfos <<- tail(.pendingFileInfos, -1)
 
-      filename <- file.path(.dir, as.character(length(.files$name)))
+      basename <- .currentFileInfo$name
+      filename <- file.path(.dir, sanitizeFileName(basename, as.character(length(.files$name))))
```

Comment:
Possibly, I think it depends on why we'd want to impose a particular limit. I chose 255 based on this from OWASP:

> Limit the filename length. For instance, the maximum length of the name of a file plus its extension should be less than 255 characters (without any directory) in an NTFS partition.

...which I think applies only to the basename, not including separators.